src/Theme/ThemeProvider.tsx

```diff
@@ -0,0 +1,69 @@
+import { MuiThemeProvider } from '@material-ui/core/styles';
+import {
+  ExtensionMessageType,
+  sendMessageToExtension,
+  ThemeChangeMessage,
+} from '../extensions';
+import React, { useEffect } from 'react';
+import Helmet from 'react-helmet';
+
+import Baseline from './Baseline';
+import { createTheme } from '../createSaleorTheme';
+import { dark, light } from './themes';
+import { Themes } from './types';
+import { ThemeContext } from './context';
+
+export interface ThemeProviderProps {
+  isDefaultDark?: boolean;
+  overrides?: Partial<Themes>;
+}
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({
+  children,
+  isDefaultDark,
```

Comment:
I'd go for a more explicit naming here, we have 2 themes right now but it's not impossible there'll be more than 2 in the future.
```suggestion
  theme = "light",
```